src/utilstrencodings.cpp

```diff
@@ -229,7 +229,7 @@ vector<unsigned char> DecodeBase64(const char* p, bool* pfInvalid)
 string DecodeBase64(const string& str)
 {
     vector<unsigned char> vchRet = DecodeBase64(str.c_str());
-    return (vchRet.size() == 0) ? string() : string((const char*)&vchRet[0], vchRet.size());
+    return std::string((const char*)vchRet.data(), vchRet.size());
```

Comment:
Just to be complete: this is *not* the same as `std::string(vchRet.begin(), vchRet.end())` in the general case. When creating an iterator range, if the type of vch is different from the type of string, a per-element cast will happen. So `string.size()` will be `vch.size()`. Usual issues with casting between signed and unsigned apply.

`std::string((const char*)vchRet.data(), vchRet.size())` however just return the memory area with all the elements of the vector as a string. So `string.size()` will be `vch.size()*sizeof(vch::value_type)`. It is likely also more efficient.